### DISCUSSION TITLE
Admin session lifetime not working.

### DISCUSSION THREAD

**Discussion by Gerardv514** [dani-garcia/vaultwarden#5114] [General]
Running latest and admin session lifetime doesn’t appear to respect the 10 mins that I have set. Yesterday admin page was still active after a few hours changes were also able to be saved. 

        /// Admin session lifetime |> Set the lifetime of admin sessions to this value (in minutes).
        admin_session_lifetime:        i64, true,  def, 20;

**Comment by BlackDex** [comment]
How did you configured it?
Are you using a config.json which overrides your settings?
Also the default is 20 min.

What happens if you set it to 1?

**Reply by Gerardv514** [reply]
Configured through /admin page and manually changed to 10.  Prior to posting here I had set it to 1 to double check and after 5 mins it was the same result, I was still able to make changes and the page was active. 

Maybe to clarify here, I open a tab and then leave that tab open expecting it to time out, but it doesn’t. 

Or does this session timeout mean if I close the tab without logging out I can still access within that time period if I were browse back to the /admin page.

**Reply by stefan0xC** [reply]
Did you logout when testing this? 

If I set it to 1 minute (`ADMIN_SESSION_LIFETIME=1`), logout and back in, the cookie expires after 1 minute:
```
[2024-10-20 16:30:49.916][request][INFO] POST /admin/config
[2024-10-20 16:30:49.916][vaultwarden::api::admin::_][WARN] Request guard `AdminToken` failed: "Unauthorized".
[2024-10-20 16:30:49.916][rocket::server::_][WARN] Responding with registered (admin_login) /admin 401 catcher.
[2024-10-20 16:30:49.916][vaultwarden::api::admin][ERROR] Authorization failed.
[2024-10-20 16:30:49.917][response][INFO] (post_config) POST /admin/config => 401 Unauthorized
```

If I don't log out the cookie with the previous time is still valid as the expiration date was set to 20 minutes from now. If it still works after hours then something else must be wrong.

**Reply by Gerardv514** [reply]
No I didn’t log out, that was the purpose of testing the session lifetime. 

I will test another browser, the aforementioned was using iOS google chrome browser.

**Reply by Gerardv514** [reply]
Worked on iOS safari. 

So far the issue is seen in the iOS chrome browser

**Reply by stefan0xC** [reply]
Well, the encoded JWT should still have expired server-side even if the cookie has not been deleted by the client. Can you still reproduce this with the iOS chrome browser? Or did you have configured a larger value for the `ADMIN_SESSION_TIMEOUT` at some point?

**Reply by Gerardv514** [reply]
Default was 20, I changed it to 10 and there have never been any other changes or values until now (set to 1), when I noticed admin page was still active 2 hrs after having been in it. 

I did a browser cache delete and it is working again. Will see how long this last, although I must say this is alarming that this happened and who knows for how long. I’ve been banking for 4 decades and a banks site/service never once did this.

**Reply by BlackDex** [reply]
Must be a Browser bug I think. Since all we do is set a cookie lifetime, and the JWT lifetime. Which both should expire.

**Reply by stefan0xC** [reply]
@BlackDex How would that be possible? I mean the JWT is created and handled by the server? The request guard should fail if the JWT has been expired, so a buggy client should not be able to extend the lifetime somehow. (Only thing that might influence the expiration date is a misconfigured server time but even then, I'm not sure how this would affect one client only.)

**Reply by BlackDex** [reply]
If it show how keeps sending request, or auto-login that might be a thingy.

**Reply by Gerardv514** [reply]
And this browser thing would be limited to just causing an issue with Vaultwarden? If it were a browser thing wouldn’t the hundreds of other sites that have logins and timeouts see the issue too?

**Reply by BlackDex** [reply]
Well, it can't work on all others and not on just one specific browser/system. That is statistically not something which can happen.
Since the features we use are basic HTTP/Browser features, and nothing changed on that part for a long time, it just can't.
